src/Adapter/Order/CommandHandler/UpdateOrderShippingDetailsHandler.php

```diff
@@ -68,59 +75,66 @@ public function handle(UpdateOrderShippingDetailsCommand $command)
         $carrierId = $command->getNewCarrierId();
         $oldTrackingNumber = $order->shipping_number;
 
-        $orderCarrier = new OrderCarrier($command->getCurrentOrderCarrierId());
-        if (!Validate::isLoadedObject($orderCarrier)) {
-            throw new OrderException('The order carrier ID is invalid.');
-        }
+        $this->contextStateManager
+            ->setTranslatorLocale((new Language($order->id_lang))->locale);
```

Comment:
Are we sure that
```
new Language($order->id_lang))
```
will return a valid Language object, so we can fetch `->locale`?